File: go/space-age/space_age.go
// Package space calculates the age of each planet relative it's own orbital period.
package space

type Planet string

const earthOrbialPeriod = 31557600

var ORBITAL_PERIODS = map[Planet]float64 	{
	Planet("Earth"): earthOrbialPeriod,
	Planet("Mercury"): earthOrbialPeriod * 0.2408467,
	Planet("Venus"): earthOrbialPeriod * 0.61519726,
	Planet("Mars"): earthOrbialPeriod * 1.8808158,
	Planet("Jupiter"): earthOrbialPeriod * 11.862615,
	Planet("Saturn"): earthOrbialPeriod * 29.447498,
	Planet("Uranus"): earthOrbialPeriod * 84.016846,
	Planet("Neptune"): earthOrbialPeriod * 164.79132,
}

// Age calculates the age of a certain amount of seconds for a given planet.
func Age(seconds float64, planet Planet) float64 {
	return seconds / ORBITAL_PERIODS[planet]
}

File: java/space-age/src/main/java/SpaceAge.java
class SpaceAge 
{
    private double seconds;

    SpaceAge(double seconds) 
    {
        this.seconds = seconds;
    }

    double onEarth() 
    {
        return convertAgeTo(Planet.EARTH);
    }

    double onMercury() 
    {
        return convertAgeTo(Planet.MERCURY);
    }

    double onVenus() 
    {
        return convertAgeTo(Planet.VENUS);
    }

    double onMars() 
    {
        return convertAgeTo(Planet.MARS);
    }

    double onJupiter() 
    {
        return convertAgeTo(Planet.JUPITER);
    }

    double onSaturn() 
    {
        return convertAgeTo(Planet.SATURN);
    }

    double onUranus() 
    {
        return convertAgeTo(Planet.URANUS);
    }

    double onNeptune() 
    {
        return convertAgeTo(Planet.NEPTUNE);
    }

    private double convertAgeTo(Planet planet)
    {
        return this.seconds / (planet.yearInEarthYears() * Planet.EARTH_YEAR_SECS);
    }

    double getSeconds()
    {
        return seconds;
    }
}

enum Planet 
{
    EARTH(1d),
    MERCURY(0.2408467),
    VENUS(0.61519726),
    MARS(1.8808158),
    JUPITER(11.862615),
    SATURN(29.447498),
    URANUS(84.016846),
    NEPTUNE(164.79132);

    private double yearInEarthYears;

    Planet(double yearInEarthYears)
    {
        this.yearInEarthYears = yearInEarthYears;
    }

    public double yearInEarthYears()
    {
        return this.yearInEarthYears;
    }

    static final double EARTH_YEAR_SECS = 31557600d;
}
